lib/plugins/cors.js

```diff
@@ -35,16 +35,25 @@ var AC_EXPOSE_HEADERS = 'Access-Control-Expose-Headers';
 
 ///--- Internal Functions
 
+// turn a string defining an allowed domain into a regex
+function toRegExp(str) {
+    if (str instanceof RegExp) {
+        return str;
+    }
+    var regexStr = str
+      // escape all the special characters, except *
+      .replace(/[\-\[\]\/\{\}\(\)\+\?\.\\\^\$\|]/g, '\\$&')
+      // replace all instances of * with a proper regex wildcard
+      .replace(/\*/g, '.*');
+
+    return new RegExp('^(https?://)?' + regexStr + '$', 'i');
+}
+
 function matchOrigin(req, origins) {
     var origin = req.headers.origin;
 
     function belongs(o) {
-        if (origin === o || o === '*') {
-            origin = o;
-            return (true);
-        }
-
-        return (false);
+        return o.test(origin);
```

Comment:
Fair point - I think I'd still prefer the differentiation due to subtle behavior changes that may not be captured in existing tests (plus the additional regex overhead). But I don't feel particularly strongly on this issue. @yunong, thoughts?